app/src/main/java/net/squanchy/schedule/SchedulePageView.kt

```diff
@@ -120,25 +124,56 @@ class SchedulePageView @JvmOverloads constructor(
     }
 
     fun updateWith(schedule: Schedule, onEventClicked: (Event) -> Unit) {
-        viewPagerAdapter.updateWith(schedule.pages, onEventClicked)
+        val initialEventForPage = schedule.pages.map(::findNextEventForPage).toTypedArray()
+        viewPagerAdapter.updateWith(schedule.pages, initialEventForPage, onEventClicked)
+
+        val todayPageIndex = findTodayIndexOrDefault(schedule.pages)
+        viewpager.setCurrentItem(todayPageIndex, false)
+
+        tabstrip.addOnTabSelectedListener(ScrollingOnTabSelectedListener(schedule, viewPagerAdapter))
         progressbar.visibility = View.GONE
     }
 
-    private class TrackingOnTabSelectedListener constructor(
-            private val analytics: Analytics,
+    private fun findTodayIndexOrDefault(pages: List<SchedulePage>): Int {
+        val now = currentTime.currentLocalDateTime()
+        return pages.firstOrNull { it.date.toLocalDate().isEqual(now.toLocalDate()) }
```

Comment:
instead of doing 2 cycles (`firstOrNull` and then `indexOf`), you can use `indexOfFirst`with this predicate, then defaulting to 0 if the result is -1

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/index-of-first.html

Reply:
possible clean implementation can be:

```kotlin
return pages.IndexOfFirst { ... }.let {
  when(it) {
    -1 -> 0
    else -> it
  }
}
```